File: PrismaApi/PrismaApi.Domain/Dtos/UncertaintyDtos.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace PrismaApi.Domain.Dtos;

public class UncertaintyDto
{
    [JsonPropertyName("id")]
    public Guid Id { get; set; } = Guid.NewGuid();
    [JsonPropertyName("issue_id")]
    public Guid IssueId { get; set; }
    [JsonPropertyName("is_key")]
    public bool IsKey { get; set; } = true;
}

public class UncertaintyIncomingDto : UncertaintyDto
{
    [JsonPropertyName("outcomes")]
    public List<OutcomeIncomingDto> Outcomes { get; set; } = new();
    public List<DiscreteProbabilityDto> DiscreteProbabilities { get; set; } = new();
}

public class UncertaintyOutgoingDto : UncertaintyDto
{
    [JsonPropertyName("outcomes")]
    public List<OutcomeOutgoingDto> Outcomes { get; set; } = new();
    [JsonPropertyName("discrete_probabilities")]
    public List<DiscreteProbabilityDto> DiscreteProbabilities { get; set; } = new();
}

File: PrismaFastApi/src/dtos/uncertainty_dtos.py
import uuid
from pydantic import BaseModel, Field
from typing import List

from src.dtos.outcome_dtos import (
    OutcomeIncomingDto,
    OutcomeOutgoingDto,
)


class UncertaintyDto(BaseModel):
    id: uuid.UUID = Field(default_factory=uuid.uuid4)
    issue_id: uuid.UUID
    is_key: bool = True


class UncertaintyIncomingDto(UncertaintyDto):
    outcomes: List[OutcomeIncomingDto] = []


class UncertaintyOutgoingDto(UncertaintyDto):
    outcomes: List[OutcomeOutgoingDto] = []
